### ISSUE TITLE
Problem installing V8 in R 3.6.2 on Arch Linux

### ISSUE THREAD

**Issue by matguidi** [jeroen/V8#80]
Hi,
I'm trying to install V8 and installation fails with the following information:

'#
'# Fatal error in , line 0
'# Failed to deserialize the V8 snapshot blob. This can mean that the snapshot blob file is corrupted or missing.
'#
'#
'#
'#FailureMessage Object: 0x7fffa8ad6630
==== C stack trace ===============================

    /lib/libv8_libbase.so(v8::base::debug::StackTrace::StackTrace()+0x16) [0x7fe0627f2386]
    /lib/libv8_libplatform.so(+0x9c4c) [0x7fe062804c4c]
    /lib/libv8_libbase.so(V8_Fatal(char const*, ...)+0x179) [0x7fe0627ed5c9]
    /lib/libv8.so(+0x23d2af) [0x7fe060e8d2af]
    /lib/libv8.so(v8::Isolate::New(v8::Isolate::CreateParams const&)+0x1e) [0x7fe060e8d2ce]
    /home/myname/R/x86_64-pc-linux-gnu-library/3.6/00LOCK-V8/00new/V8/libs/V8.so(+0xecfe) [0x7fe06281ecfe]
    /usr/lib64/R/lib/libR.so(+0x6cfcf) [0x7fe066274fcf]
    /usr/lib64/R/lib/libR.so(+0x6d2d8) [0x7fe0662752d8]
    /usr/lib64/R/lib/libR.so(+0x12eb96) [0x7fe066336b96]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x145eb0) [0x7fe06634deb0]
    /usr/lib64/R/lib/libR.so(Rf_applyClosure+0x165) [0x7fe06634ed25]
    /usr/lib64/R/lib/libR.so(+0x13830e) [0x7fe06634030e]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x145eb0) [0x7fe06634deb0]
    /usr/lib64/R/lib/libR.so(Rf_applyClosure+0x165) [0x7fe06634ed25]
    /usr/lib64/R/lib/libR.so(+0x13830e) [0x7fe06634030e]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x145eb0) [0x7fe06634deb0]
    /usr/lib64/R/lib/libR.so(Rf_applyClosure+0x165) [0x7fe06634ed25]
    /usr/lib64/R/lib/libR.so(+0x13830e) [0x7fe06634030e]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x1449af) [0x7fe06634c9af]
    /usr/lib64/R/lib/libR.so(+0x144ef4) [0x7fe06634cef4]
    /usr/lib64/R/lib/libR.so(+0x132ae3) [0x7fe06633aae3]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x1449af) [0x7fe06634c9af]
    /usr/lib64/R/lib/libR.so(+0x144ef4) [0x7fe06634cef4]
    /usr/lib64/R/lib/libR.so(+0x132ae3) [0x7fe06633aae3]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x1449af) [0x7fe06634c9af]
    /usr/lib64/R/lib/libR.so(+0x144ef4) [0x7fe06634cef4]
    /usr/lib64/R/lib/libR.so(+0x132ae3) [0x7fe06633aae3]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x1449af) [0x7fe06634c9af]
    /usr/lib64/R/lib/libR.so(+0x144ef4) [0x7fe06634cef4]
    /usr/lib64/R/lib/libR.so(+0x132ae3) [0x7fe06633aae3]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x145eb0) [0x7fe06634deb0]
    /usr/lib64/R/lib/libR.so(Rf_applyClosure+0x165) [0x7fe06634ed25]
    /usr/lib64/R/lib/libR.so(+0x13830e) [0x7fe06634030e]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x145eb0) [0x7fe06634deb0]
    /usr/lib64/R/lib/libR.so(Rf_applyClosure+0x165) [0x7fe06634ed25]
    /usr/lib64/R/lib/libR.so(+0x13830e) [0x7fe06634030e]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x145eb0) [0x7fe06634deb0]
    /usr/lib64/R/lib/libR.so(Rf_applyClosure+0x165) [0x7fe06634ed25]
    /usr/lib64/R/lib/libR.so(+0x13830e) [0x7fe06634030e]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x145eb0) [0x7fe06634deb0]
    /usr/lib64/R/lib/libR.so(Rf_applyClosure+0x165) [0x7fe06634ed25]
    /usr/lib64/R/lib/libR.so(+0x13830e) [0x7fe06634030e]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x145eb0) [0x7fe06634deb0]
    /usr/lib64/R/lib/libR.so(Rf_applyClosure+0x165) [0x7fe06634ed25]
    /usr/lib64/R/lib/libR.so(+0x13830e) [0x7fe06634030e]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]
    /usr/lib64/R/lib/libR.so(+0x1449af) [0x7fe06634c9af]
    /usr/lib64/R/lib/libR.so(+0x144ef4) [0x7fe06634cef4]
    /usr/lib64/R/lib/libR.so(+0x132ae3) [0x7fe06633aae3]
    /usr/lib64/R/lib/libR.so(Rf_eval+0x1a0) [0x7fe06634c050]

 *** caught illegal operation ***
address 0x7fe0627e4bc9, cause 'illegal operand'

Traceback:
 1: dyn.load(file, DLLpath = DLLpath, ...)
 2: library.dynam(lib, package, package.lib)
 3: loadNamespace(package, lib.loc)
 4: doTryCatch(return(expr), name, parentenv, handler)
 5: tryCatchOne(expr, names, parentenv, handlers[[1L]])
 6: tryCatchList(expr, classes, parentenv, handlers)
 7: tryCatch({    attr(package, "LibPath") <- which.lib.loc    ns <- loadNamespace(package, lib.loc)    env <- attachNamespace(ns, pos = pos, deps, exclude, include.only)}, error = function(e) {    P <- if (!is.null(cc <- conditionCall(e)))         paste(" in", deparse(cc)[1L])    else ""    msg <- gettextf("package or namespace load failed for %s%s:\n %s",         sQuote(package), P, conditionMessage(e))    if (logical.return)         message(paste("Error:", msg), domain = NA)    else stop(msg, call. = FALSE, domain = NA)})
 8: library(pkg_name, lib.loc = lib, character.only = TRUE, logical.return = TRUE)
 9: withCallingHandlers(expr, packageStartupMessage = function(c) invokeRestart("muffleMessage"))
10: suppressPackageStartupMessages(library(pkg_name, lib.loc = lib,     character.only = TRUE, logical.return = TRUE))
11: doTryCatch(return(expr), name, parentenv, handler)
12: tryCatchOne(expr, names, parentenv, handlers[[1L]])
13: tryCatchList(expr, classes, parentenv, handlers)
14: tryCatch(expr, error = function(e) {    call <- conditionCall(e)    if (!is.null(call)) {        if (identical(call[[1L]], quote(doTryCatch)))             call <- sys.call(-4L)        dcall <- deparse(call)[1L]        prefix <- paste("Error in", dcall, ": ")        LONG <- 75L        sm <- strsplit(conditionMessage(e), "\n")[[1L]]        w <- 14L + nchar(dcall, type = "w") + nchar(sm[1L], type = "w")        if (is.na(w))             w <- 14L + nchar(dcall, type = "b") + nchar(sm[1L],                 type = "b")        if (w > LONG)             prefix <- paste0(prefix, "\n  ")    }    else prefix <- "Error : "    msg <- paste0(prefix, conditionMessage(e), "\n")    .Internal(seterrmessage(msg[1L]))    if (!silent && isTRUE(getOption("show.error.messages"))) {        cat(msg, file = outFile)        .Internal(printDeferredWarnings())    }    invisible(structure(msg, class = "try-error", condition = e))})
15: try(suppressPackageStartupMessages(library(pkg_name, lib.loc = lib,     character.only = TRUE, logical.return = TRUE)))
16: tools:::.test_load_package("V8", "/home/myname/R/x86_64-pc-linux-gnu-library/3.6/00LOCK-V8/00new")
An irrecoverable exception occurred. R is aborting now ...

I run R version 3.6.2 in Manjaro Linux. I have successfully compiled and installed the V8 package in my system. I have tried to install both with the standard procedure
`install.packages("V8")`
and from github
`install_github("jeroen/v8")`
The error comes up exactly the same.

**Comment by jeroen** [comment]
Which version of libv8 is your system running? Did you have the same problem with the previous v8? Can you try this:


```r
install.packages("https://cran.r-project.org/src/contrib/Archive/V8/V8_2.3.tar.gz", repos = NULL)
```

**Comment by matguidi** [comment]
I am running version 7.9.317.32-1 (https://aur.archlinux.org/packages/v8). I can't tell you how previous versions of v8 behave because I have just installed it for the first time. 

Trying to install the version that you suggest gives the same error.

**Comment by matguidi** [comment]
I am currently install this version: https://aur.archlinux.org/packages/v8-r/. I'll report if it solves the problem.

**Comment by matguidi** [comment]
I confirm installing v8-r solves the problem

**Comment by jeroen** [comment]
OK great, you can thank @JanMarvin

**Comment by grantmcdermott** [comment]
Was just about to post a similar comment after switching to the v8-r build. 

@jeroen do you mind adding a note to the README, advising Arch users to consider Jan's v8-r version instead of the standard v8 one? Happy to add that myself if you'd rather just accept a PR.

**Comment by jeroen** [comment]
Yes, I have asked the v8-r author the same thing before: https://github.com/jeroen/V8/pull/71#issuecomment-536274125

I don't use Arch linux myself, but if you send a PR with instructions that make sense, I'll merge it.

**Comment by JanMarvin** [comment]
Thanks, forgot it. If you have a pr ready @grantmcdermott that'll be fine. Otherwise I'll send one in tomorrow

**Comment by jeroen** [comment]
Is there anything we can do to vote for your `v8-r` becoming the default `v8` package in arch? It's so strange that the default `v8` is completely broken while yours works great.

**Comment by JanMarvin** [comment]
Not that I'm aware of. Both packages are in the aur which is the user repository containing only pkgbuilds and no binaries. In theory aur packages that are voted for might end up in the arch binary repository. Though I'm not familiar with the process. Arch only provides a (compared to Debian or Red Hat based distributions) limited amount of pre build packages and even widely used software such as rstudio is not yet available outside of the aur. That's kinda the Arch philosophy ¯\_(ツ)_/¯

**Comment by jeroen** [comment]
So can we somehow suggest your PKGBUILD to be incorporated by the `v8` package? There is nothing specific for R in your v8-r, I'm sure the arch v8 package crashes with bindings from other languages as well...

**Comment by JanMarvin** [comment]
That was my initial idea and I suggested it in the comment section of the package, but the maintainer of the aur v8 package sadly never picked it up. Though he added a comment linking to v8-r when I pushed it to the aur a few weeks later. Since his pkgbuild is still maintained and obviously works for whatever he's doing with it, I don't see anything we can do aside from documenting which package to use.
